plugins/src/main/java/com/google/fhir/gateway/plugin/OpenSRPSyncAccessDecision.java

```diff
@@ -100,10 +107,21 @@ public RequestMutation getRequestMutation(RequestDetailsReader requestDetailsRea
     //  This does not bar access to anyone who uses their own sync tags to circumvent
     //  the filter. The aim of this feature based on scoping was to pre-filter the data for the user
```

Comment:
This might still be valid unless we remove any previous tags added as search params

Reply:
I thought your concern of a security vulnerability was which practitioner details were used by the gateway and like if the user can inject their own fake ones to bypass.

Sounds though like your concern was on them adding extra sync strategy search params to the request. I think that is okay because:
1. We overwrite any already existing query params overridden by the request mutation
2. If they did add any extra ones, it will result into an boolean AND meaning it will apply both filters

Reply:
If the user/device makes the request with search params, this results in an OR request especially if the search param is `_tag`. This allows them to get more data than they are allowed to.
We can test this out to confirm

Reply:
Looks like it is because of [this block here](https://github.com/opensrp/fhir-gateway/blob/main/plugins/src/main/java/com/google/fhir/gateway/plugin/OpenSRPSyncAccessDecision.java#L143-L147), once we remove it we should be good.